flink-table/flink-table-common/src/main/java/org/apache/flink/table/types/utils/ValueDataTypeConverter.java

```diff
@@ -126,6 +126,10 @@ private static DataType convertToDecimalType(BigDecimal decimal) {
             // negative scale is not supported, normalize it
             return DataTypes.DECIMAL(precision - scale, 0);
         }
+
+        if (scale > precision) {
+            return DataTypes.DECIMAL(precision + scale, scale);
```

Comment:
This seems to have some issues, for example, with numbers like `0.011` and `0.111`.

Reply:
it will not, for example 0.111 java BigDecimal precision is 3 and scale is also 3, will not throw eception @dylanhz 
if we change to decimal(4,3) it will cause breaking change @dylanhz

Reply:
make sense, precision must be greater than scale according to https://github.com/apache/flink/blob/6633719c728526a08344645e0a844f27d25629f4/flink-table/flink-table-common/src/main/java/org/apache/flink/table/types/logical/DecimalType.java#L74

Reply:
I agree that we should keep `scale <= precision` unchanged.

However, my main question is about `precision + scale` part. For example, as mentioned in my previous comment, the BigDecimal value `0.011` has a precision of 2 and a scale of 3, while its corresponding Flink type should be `Decimal(3, 3)`. Perhaps we should use `precision = max(precision, scale)`, as in [this hotfix](https://github.com/apache/flink/pull/2498).